### ISSUE TITLE
Instructions aren't right

### ISSUE THREAD

**Issue by Swader** [blockmason/web3-provider-ledger#10]
```js
console.log("Ledger mode");
const device = new LedgerDevice({
				"accountIndex": 0,
				"appId": location.protocol+'//'+location.hostname+(location.port ? ':'+location.port: ''),
				"u2f": u2f 
			});

const web3ledger = new window.Web3(new LedgerProvider({device}));
console.log(web3ledger);

let method = (duration == 1) ? 'monthlyPrice' : 'annualPrice';
web3net.eth.contract(abi).at(address)[method](async function(error, result) {

	console.log("Successfully called contract method");
	console.log(error);
	console.log(result);

	if (error) {
		alert("Could not fetch price");
		return false;
	}

	let tx = await web3ledger.eth.contract(abi).at(address).subscribeMe(duration, web3net.fromAscii(email), {
		value: result.toString(10), // BigNumber object
		gas: 90000
		});

	web3net.sendRawTransaction(tx, function(e, r) {
		console.log(e);
		console.log(r);
	});
});
```

Following your instructions, vaguely, will produce OK results in terms of the Ledger device and fetching its accounts. I will even be able to get to the point of `let tx = await...` but that is where it will fail because the a Ledger-provided web3 cannot find a contract `at` any address, since it has no way of connecting to the node where it should find it.

The readme says this will produce a raw transaction I should pass into the net-enabled web3, which is at the end of my snippet, but getting the raw transaction of a contract function's execution is impossible unless the contract is read from a deployed address.

What am I missing here?

**Comment by Swader** [comment]
Also note that a new instantiation of LedgerDevice requires u2f, accountIndex, and appId to be passed in - it does not use any defaults, so this is where the README is incorrect too.

**Comment by canterberry** [comment]
Hi @Swader, sorry for the long-delayed response. I've adjusted my notification settings to do a better job of making sure I see issues like this as they come in.

You're right, the README shows `LedgerDevice` as having no required parameters, which contradicts the [API reference documentation](https://14-115747070-gh.circle-artifacts.com/0/home/project/project/docs/web3-provider-ledger/1.0.6/LedgerDevice.html). If constructing a `LedgerDevice` directly, both `appId` and `u2f` need to be provided (`accountIndex` will default to zero).

Another thing the README doesn't make clear is that the Ledger-enabled web3 instance needs to be treated the same as a network-enabled instance would normally be treated -- in order to successfully sign a transaction, it still needs to be given parameters like gas price, gas limit, and nonce. The Ledger provider does, in response to a contract function call, return a signed transaction that can be handed off directly to the network-enabled web3 instance's `#sendRawTransaction()`.

After writing this provider, I discovered [ethjs-provider-signer](https://github.com/ethjs/ethjs-provider-signer), which greatly simplifies the complexity of this multi-provider strategy. That, combined with the `LedgerDevice` from this library, should make for a much cleaner implementation:

```javascript
import LedgerDevice from 'web3-provider-ledger/ledger-device';
import SignerProvider from 'ethjs-provider-signer';
import Web3 from 'web3';
import encodeTransaction from 'web3-provider-ledger/rpc/encode-transaction';

const ledgerDevice = new LedgerDevice({ appId: origin, u2f });

const provider = new SignerProvider('https://ropsten.infura.io', {
  signTransaction: (rawTransaction, callback) => {
    const signature = await device.sign(encodeTransaction(rawTransaction));
    const signedTransaction = encodeTransaction({
      ...rawTransaction,
      ...signature
    }).toString('hex');
    callback(null, signedTransaction);
  },
  accounts: async (callback) => {
    const accounts = await ledgerDevice.listAddresses();
    callback(null, accounts);
  }
});

const web3 = new Web3(provider);

// web3 is fully wired up to the Ledger device and is now ready to be used.
```

The above code *should* work right now. After going through this exercise, it's pretty clear that rolling the transaction encoding into the device would be a notable improvement in a future version to simply this even further.
